idanalysis/eqparams.py

```diff
@@ -0,0 +1,1194 @@
+        i2y = cumtrapz(dx=dx, y=(bx / self.brho) ** 2)[-1]
+        self._i2x = i2x
+        self._i2y = i2y
+        return i2x, i2y
+
+    def calc_i3(self):
+        """Calculate third radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I3x
+            float: delta I3y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        i3x = cumtrapz(dx=dx, y=_np.abs((by / self.brho) ** 3))[-1]
+        i3y = cumtrapz(dx=dx, y=_np.abs((bx / self.brho) ** 3))[-1]
+        self._i3x = i3x
+        self._i3y = i3y
+        return i3x, i3y
+
+    def calc_i4(self):
+        """Calculate fourth radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I4x
+            float: delta I4y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        dby = _np.diff(by)
+        dbx = _np.diff(bx)
+        px = cumtrapz(dx=dx, y=by) / self.brho
+        py = cumtrapz(dx=dx, y=-bx) / self.brho
+        kx = -(dby / dx) * px / self.brho
+        ky = -(dbx / dx) * py / self.brho
+        etax, etay = self.calc_dispersion()
+        i4x = cumtrapz(
+            dx=dx,
+            y=(
+                etax * (by[1:-1] / self.brho) ** 3
+                - 2 * ky[:-1] * etax * (by[1:-1] / self.brho)
+            ),
+        )[-1]
+        i4y = cumtrapz(
+            dx=dx,
+            y=(
+                etay * (bx[1:-1] / self.brho) ** 3
+                - 2 * kx[:-1] * etay * (bx[1:-1] / self.brho)
+            ),
+        )[-1]
+        self._i4x = i4x
+        self._i4y = i4y
+        return i4x, i4y
+
+    def calc_i5(self):
+        """Calculate fifth radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I5x
+            float: delta I5y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        hx, hy = self.get_curly_h()
+        i5x = cumtrapz(
+            dx=dx, y=hx * _np.abs((by / self.brho) ** 3)[1:-1]
+        )[-1]
+        i5y = cumtrapz(
+            dx=dx, y=hy * _np.abs((bx / self.brho) ** 3)[1:-1]
+        )[-1]
+        self._i5x = i5x
+        self._i5y = i5y
+        return i5x, i5y
+
+
+class EqParamAnalysis:
+    """Class to calculate beam equilibrium parameters.
+
+    All equations used here are from the book: Clarke, James A.,
+    'The Science and Technology of Undulators and Wigglers'.
+    Chapter 10: 'The Effect of Insertion Devices on the Electron Beam'.
+    """
+
+    def __init__(self):
+        """Class constructor."""
+        self.ids = list()
+        self._model = pymodels.si.create_accelerator()
+        self._eq_params_nominal = None
+        self._beam = Beam(3)
```

Comment:
I think the energy could be an optional class parameter with default 3 GeV. what is SIRIUS energy, 3 ou 2.98 GeV ? what is the effect of this small discrepancy on EqParameters ? an optional argument can be used to answer this question.